### ISSUE TITLE
Removing attribute from family keep generating warnings

### ISSUE THREAD

**Issue by nei** [akeneo/pim-community-dev#7906]
I've removed an attribute from all the families, then I've deleted the attribute. I was expecting to have this attribute gone but when I am running import/export/reindex I get loads of warnings.

Akeneo: CE 2.1.0
PHP 7.1.12
Mysql 5.7.20

```
12:48:22 WARNING   [app] Tried to load a product value with the attribute "thumbnail" that does not exist. [] []
12:48:22 WARNING   [app] Tried to load a product value with the attribute "thumbnail" that does not exist. [] []
```

Is this an issue? How can I clean my product data to get rid of these warnings?

**Comment by damien-carcel** [comment]
Hello @nei.

This warning is perfectly normal, and it is really just info for developers.

When removing an attribute (or an attribute option), the corresponding values in the database, in the JSON raw_value field of the product table, are not purged. This is a wanted behavior to prevent the execution of a very long-cleaning process that could harm the PIM if not handle correctly.

Instead, the factory that is in charge of created real PHP objects from the JSON raw values will check if attributes (or options, or locales, channels, etc.) still exists. If it doesn't, then it throws the warning you saw. But next time you'll save one of those products, as the value was not hydrated, it will be removed from the raw_values and the warning will disappear. Eventually, after updating all the concerned products, all warning will be gone.

So to sum up, you can ignore those warnings, they are harmless. But if you really want to get rid of them, you could just create a command that iterates on your products and update them without any modifications. This should remove the useless values.

Hope this helps.

Regards,
Damien.

**Comment by damien-carcel** [comment]
Sorry I closed the issue, I press "Close and comment" instead of just comment :confounded:.

**Comment by nei** [comment]
Right, thank you for all this detailed explanation it makes sense. @damien-carcel
